command/genesis/polybft_params.go

```diff
@@ -125,24 +123,41 @@ func (p *genesisParams) generatePolyBftChainConfig(o command.OutputFormatter) er
 		return err
 	}
 
-	// either premine non-validator or override validator accounts balance
-	for _, premine := range p.premine {
-		premineInfo, err := parsePremineInfo(premine)
+	premineInfos := make([]*PremineInfo, len(p.premine))
+	premineValidatorsAddrs := []string{}
+	// premine non-validator
+	for i, premine := range p.premine {
+		premineInfo, err := ParsePremineInfo(premine)
 		if err != nil {
 			return err
 		}
 
-		if i, ok := validatorPreminesMap[premineInfo.address]; ok {
-			premineInfos[i] = premineInfo
-		} else {
-			premineInfos = append(premineInfos, premineInfo) //nolint:makezero
+		// collect validators addresses which got premined, as it is an error
+		// genesis validators balances must be defined in manifest file and should not be changed in the genesis
+		if _, ok := genesisValidators[premineInfo.Address]; ok {
+			premineValidatorsAddrs = append(premineValidatorsAddrs, premineInfo.Address.String())
+		}
+
+		premineInfos[i] = premineInfo
```

Comment:
I think this should only be added to the list if it is not in `genesisValidators` map. Right?

Reply:
Yes you are right to some extent, although if `premineValidatorsAddrs` is non-empty, we are returning an error anyways (https://github.com/0xPolygon/polygon-edge/pull/1314/files#diff-5ffa938067dd887484ee4d986923c49d71fdb71f0aeacd481518cfdf8e3475a1R145-R148).

I'm going to change it if it makes more sense to you.